src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -460,76 +438,66 @@ private ObjectMetadata syncPutObject(
     ) throws NoSuchAlgorithmException, PidRefsFileExistsException, IOException, RuntimeException,
         InterruptedException {
         // Lock pid for thread safety, transaction control and atomic writing
-        // A pid can only be stored once and only once, subsequent calls will
-        // be accepted but will be rejected if pid hash object exists
+        // An object is stored once and only once
```

Comment:
May we put the ` synchronized (objectLockedIds) {` block into the following `try {` block? So far there is not issue. But in the future, if someone add the lines between the block and `try {`, it can cause issue.

Reply:
Thank you! Sorry about missing this.